packages/sdk-provider-solana/src/actions/sendAndConfirmTransaction.ts

```diff
@@ -86,28 +105,40 @@ export async function sendAndConfirmTransaction(
           }
 
           await sleep(400)
+
+          if (!abortController.signal.aborted) {
+            blockHeight = await connection
+              .getBlockHeight({
+                commitment: 'confirmed',
+              })
+              .send()
```

Comment:
We didn't have a block height update here to reduce the number of requests to the RPC. I think it is fine to update it every second for both cases when we send a transaction below. Wdyt?

Reply:
Hmm, I separated them to avoid race conditions as we are sharing `blockheight` between two loops.
The polling loop is updating the blockheight, while the sending loop is reading it.

Yes, we get 2x more rpc calls, but better code quality.

Reply:
It does make sense to update once every second when resending, Since the worst case scenario is minimal during a race condition, it just feels dirty.